SvmTask.py

```diff
@@ -0,0 +1,565 @@
+
+
+val_range = [ j for j in range(0, numFolds) ]
+split_range =  [ 1.0/numFolds for j in range(0, numFolds)]
+gl_valid_err = 2
+best_i = 0
+best_j = 0
+best_k = 0
+
+split_data = labeled_df.randomSplit(split_range)
+split_train_data = []
+split_val_data = []
+for t in val_range:
+    split_val_data.insert(t, split_data[t])
+    for v in val_range:
+        if t!=v:
+            if len(split_train_data) > t:
+                split_train_data[t] = split_train_data[t].unionAll(split_data[v])
+            else:
+                split_train_data.insert(t, split_data[v])
+
+print 'data prepared'
+
+for i in my_iterations:
+    for j in my_regParam:
+        for k in my_step:
+            validErr = 0
+            for v in val_range:
+                svmmodel = SVMWithSGD.train(split_train_data[v].select(col("label"), col("feature_vector")).map(lambda row: LabeledPoint(row.label, row.feature_vector)),
+                                    iterations=i, regParam=j,step=k)
+                valid_pred = split_val_data[v].map(lambda p: (p.label, svmmodel.predict(p.feature_vector)))
+                validErr += valid_pred.filter(lambda (a, p): a != p).count() / float(valid_pred.count())
+            validErr = validErr/ numFolds
+            print 'Iterations : ' + str(i) + ' RegParam : ' +   str(j) +  ' Step size : ' + str(k) + ' Validation Error : ' + str(validErr)
+            if validErr < gl_valid_err:
+                gl_valid_err = validErr
+                best_i = i
+                best_j = j
+                best_k = k
+
+svmmodel = SVMWithSGD.train(labeled_df.select(col("label"), col("feature_vector")).map(lambda row: LabeledPoint(row.label, row.feature_vector)),
+                            iterations=best_i, regParam=best_j, step=best_k)
+'''
+svm = SVMWithSGD.train(labeled_df.select(col("label"), col("feature_vector")).map(lambda row: LabeledPoint(row.label, row.feature_vector)),iterations=5000, regParam=0.01,step=0.125)
+paramGrid = ParamGridBuilder().build()
+
+evaluator = BinaryClassificationEvaluator()
+
+cv = CrossValidator(estimator=pipeline, estimatorParamMaps=paramGrid, evaluator=evaluator, numFolds=5)
+cvmodel = cv.fit(labeled_df)
+'''
+
+print('Evaluation ....')
+
+# Evaluating the model on training ddf_feats = read_from_cassandra(sqlContext, my_keyspace, features_table_name)ata
+train_pred = labeled_df.map(lambda p: (p.label, svmmodel.predict(p.feature_vector)))
+trainErr = train_pred.filter(lambda (a, p): a != p).count() / float(train_pred.count())
+print("Training Error = " + str(trainErr))
+
+#valid_pred = validate_df.map(lambda p: (p.label, svmmodel.predict(p.feature_vector)))
+#validErr = valid_pred.filter(lambda (a, p): a != p).count() / float(valid_pred.count())
+#print("Validation Error = " + str(validErr))
+
+def generate_uuid():
+    return str(uuid.uuid4())
+svmmodel.clearThreshold()
+testrow = test_df.map(lambda p: (generate_uuid(), p.image_id, svmmodel.predict(p.feature_vector).tolist())).collect()
```

Comment:
this should be a single UUID for the whole results set. But that's probably my bad communication.